src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            array = new Object[capacity];
+        } else {
+            throw new IllegalArgumentException("Capacity must be greater than zero.");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= array.length) {
+            grow();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Non existent "
+                    + "position: " + index);
```

Comment:
The custom exception `ArrayListIndexOutOfBoundsException` is referenced here but is not defined in the code. Please ensure this class is implemented to handle invalid index cases as specified in the task requirements.

Reply:
It is in separate class